twofactor_privacyidea/lib/Provider/TwoFactorPrivacyIDEAProvider.php

```diff
@@ -364,7 +364,7 @@ public function authenticate($username, $password) {
                 $options['body']["signaturedata"] = $signatureData;
                 $options['body']["clientdata"] = $clientData;
             }
-        } else {
+        } else if ($pushResponse || $tiqrResponse) {
```

Comment:
Thank you for this fix, but what happens now if the user enters an empty password? Will we get an error because ``$url`` is undefined? I think in this case, it would be good to send the ``/validate/check`` request anyway. Correct me if I'm wrong, but couldn't we simply swap the two conditional branches, such that we have
```php
if ($pushResponse || $tiqrResponse) {
...
} else {
...
}
```
?
Nice branch name btw :)